enhancements/network/ip-interface-selection.yaml

```diff
@@ -0,0 +1,302 @@
+---
+title: ip-interface-selection
+authors:
+  - @cybertron
+reviewers:
+  - @jcaamano
+  - @tsorya
+approvers:
+  - TBD
+api-approvers:
+  - "None"
+creation-date: 2022-07-07
+last-updated: 2022-07-07
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - https://github.com/openshift/baremetal-runtimecfg/issues/119
+replaces:
+superseded-by:
+---
+
+# IP and Interface Selection
+
+## Summary
+
+As OpenShift is deployed in increasingly complex networking environments, we
+have gotten many requests for more control over which interface is used for
+the primary node IP. We provided a basic mechanism for this with
+[KUBELET_NODEIP_HINT](https://github.com/openshift/machine-config-operator/pull/2888)
+but as users have started to exercise that some significant limitations have
+come to light.
+
+## Motivation
+
+Some users want to have a great deal of control over how their network traffic
+is routed. Because we use the default route for interface and IP selection,
+in some cases they are not able to route traffic the way they want.
+
+### User Stories
+
+As a network administrator, I want cluster traffic to be on one NIC and the
```

Comment:
* As an example BZ where users might benefit from the feature: https://bugzilla.redhat.com/show_bug.cgi?id=2064131

* We have customers with multiple NICs attached to their servers and they want control over which NIC to use.

* It seems that this is a very common and important need for SNO deployments.

* I wouldn't be surprised if this was important for OpenStack deployments with manila (= 2 interfaces)

Reply:
@tsorya @rbbratta might help fill in the user story?
Also: https://coreos.slack.com/archives/CDCP2LA9L/p1652189014050609

Reply:
In SNO deployment with 2 or more nics  , we want to be sure that configure-ovs and nodeip-configuration chooses same ip, the one that user asked for, if not installation fails as router is not reachable

Reply:
There was a customer request for dual-stack to separate IPv4 and IPv6 on different NICs, so we need to be aware of dual-stack issues.

Reply:
Hmm, separating dual stack IPs may be out of scope for this iteration, but we should keep it in mind. That's probably another argument in favor of having this work based solely on IP and not have the interface file at all.